src/CharSetView.cpp

```diff
@@ -266,8 +269,6 @@ void CharSetView::MouseDown(BPoint point)
 
 void CharSetView::KeyDown(const char *bytes, int32 numBytes)
 {
-	uint16		uniChar[1];
-
 	if (numBytes == 1) {
```

Comment:
No, that wouldn't work. The Tab key is just one example. However, we can do the conversion at the beginning of the function and then use the converted 32-bit value or code point in the switch. It would a bit slower, although insignificantly.

Reply:
You should keep the check that you already have in the default for bytes[0]

Reply:
I understand that's what you suggested, but if bytes[0] is an ASCII code and numBytes is not 1, then the input is not a valid UTF-8 encoding.

Reply:
if it is an ASCII code it is 1.
http://xref.plausible.coop/source/s?n=25&start=0&sort=relevancy&q=KeyDown&project=haiku
for example http://xref.plausible.coop/source/xref/haiku/src/kits/tracker/PoseView.cpp#6584

I stop try to give advice...

Reply:
I wasn't clear. What I meant was that looking at `bytes[0]` alone can't determine if `bytes` is a valid UTF-8 encoding. For example, if `numBytes = 2` and `bytes[0]` = 9 (and `bytes[1]` = whatever), we don't have a valid UTF-8 encoding.